ccextender.yaml

```diff
@@ -1,44 +1,316 @@
+                      \nMICROS_HACK_ARTIFACT_GW_OUTBOUND := docker.atl-paas.net$(IMAGE_PATH)/$(GATEWAY_OUTBOUND_IMAGE_NAME)"
+      makefile_hack_deploy: "
+                             \   BENTHOS_OUTPUT_ARTIFACT_NAME=$(MICROS_HACK_ARTIFACT_BENTHOS_OUTPUT) \\\n
+                             \   GW_OUTBOUND_ARTIFACT_NAME=$(MICROS_HACK_ARTIFACT_GW_OUTBOUND) \\"
+  streamer-manager:
+    template-streamer-manager:
+    template-micros:
+      service_descriptor: "
+                           \ streamer-manager:\n
+                           \   image: ${SM_ARTIFACT_NAME} #docker.atl-paas.net/sox/asecurityteam/sec-awsconfig-transformerd-streamer-manager\n
+                           \   tag: ${ARTIFACT_VERSION}\n
+                           \   links:\n
+                           \     - benthos-input\n
+                           \     - statsd-service-sidecar\n
+                           \   ports:\n
+                           \     - \"8085:8085\"\n
+                           \   deployRestartCount: 3\n
+                           \ lifecycle:\n
+                           \   image: docker.atl-paas.net/sox/asecurityteam/microslifecycle\n
+                           \   tag: latest\n
+                           \   links:\n
+                           \     - streamer-manager\n
+                           \     - statsd-service-sidecar\n
+                           \   deployRestartCount: 3"
+      docker_compose: "
+                       \ streamer-manager:\n
+                       \   build:\n
+                       \     context: .\n
+                       \     dockerfile: streamer-manager.Dockerfile\n
+                       \     args:\n
+                       \       REGISTRY: ${REGISTRY}\n
+                       \   image: ${IMAGE}-streamer-manager:${TAG}\n
+                       \   ports:\n
+                       \     - 8085:8085\n
+                       \   environment:\n
+                       \     STREAMERMANAGER_HTTPSERVER_ADDRESS: \":8085\"\n
+                       \     STREAMERMANAGER_LOGGER_OUTPUT: \"STDOUT\"\n
+                       \     STREAMERMANAGER_STATS_OUTPUT: \"NULL\"\n
+                       \     STREAMERMANAGER_DIRECTORY: \"/configs\"\n
+                       \     STREAMERMANAGER_PATH: \"/lifecycle\"\n
+                       \     STREAMERMANAGER_DESTINATION: \"http://benthos-input:4195\"\n
+                       \ lifecycle:\n
+                       \   image: docker.atl-paas.net/sox/asecurityteam/microslifecycle:latest\n
+                       \   environment:\n
+                       \     MICROSLIFECYCLE_HTTP_NOTIFICATIONS: \"http://streamer-manager:8085/lifecycle\""
+    template-makefile:
+      makefile_image: "STREAMER_MANAGER_IMAGE_NAME := $(APP_IMAGE_NAME)-streamer-manager"
+      makefile_hack: "MICROS_HACK_ARTIFACT_SM := docker.atl-paas.net$(IMAGE_PATH)/$(STREAMER_MANAGER_IMAGE_NAME)"
+      makefile_hack_deploy: "\    SM_ARTIFACT_NAME=$(MICROS_HACK_ARTIFACT_SM) \\"
 locations:
```

Comment:
Question: What does `location` do?

Reply:
The locations block lets CCExtender know where it can grab the templates from. For instance, for template-docker, it looks for a directory called "template docker" at the location defined under the github block: /home/sdcli/oss-templates/, which points ccextender to: /home/sdcli/oss-templates/template-docker.

Reply:
oh ok. Is there a programatic way we can handle these? 

``` 
 local-bb:
    - /Users/aslape/templates/bitbucket-new/
  local-gh:
    - /Users/aslape/templates/github-new/
```

Reply:
Yeah, sorry about that. That's another thing I missed. Those two fields are deprecated now, so I need to take them out. Thanks for catching that.